Week1/homework 1/app.js

```diff
@@ -0,0 +1,156 @@
+        await execQuery(CREATE_TABLE_CITIES);
+        dataOfCities.forEach(city => {
+            
+            let sql1 = 'INSERT IGNORE INTO cities SET ?';
+            let query = db.query(sql1, city, (err, result) => {
+                if(err) throw err;
+            });
+            // }
+        });
+        console.log('Data imported to cities table')
+    }catch (error){
+       console.error(error);
+   }    
+}
+createCitiesTableAndInsertData()  
+
+
+
+// 4- Write queries (that's a technical term meaning 'requests for information from a database') that will retrieve data that answers the following questions:
+
+
+// Filtering (using async/await to start execute queries after create the tables and import the data )
+    async function filtering(sql){
+        await createCountriesTableAndInsertData;
+        await createCitiesTableAndInsertData;
+        let query = db.query(sql, (err, results) => {
+            if(err) throw err;
+            console.log('answer is', results);
+        });
+    }
+ 
+//  4-1 What are the names of the countries with population greater than 8 million
+let  populationGreaterThanEightMillion = 'SELECT name FROM countries WHERE population > 8000000';
+filtering(populationGreaterThanEightMillion);
+
+// 4-2 What are the names of the countries that have “land” in their names ?
+let namesHaveLand = "SELECT name FROM countries WHERE name LIKE '%land%'";
+filtering(namesHaveLand);
+
+// 4-3 What are the names of the cities with population in between 500,000 and 1 million ?
+let populationBetweenHalfAndOneMillion= 'SELECT * FROM cities WHERE Population BETWEEN 500000 AND 1000000';
+filtering(populationBetweenHalfAndOneMillion);
+
+// 4-4 What are the names of all the countries on the continent ‘Europe’ ?
+let countriesInEurope = "SELECT name FROM countries WHERE continent = 'Europe'";
+filtering(countriesInEurope);
+
+//4-5 List all the countries in the descending order based on their surface areas.
+let countriesInDescendingOrder = "SELECT name FROM countries ORDER BY surface_area DESC "
+filtering(countriesInDescendingOrder);
+
+// 5- Write queries that answer the following questions:
+
+// 5-1 What are the names of all the cities in the Netherlands?
+
+let citiesOfNetherlands = "SELECT name FROM cities WHERE country_code = 'NLD' "
+filtering(citiesOfNetherlands);
+
+// 5-2 What's the population of Rotterdam?
+let populationOfRotterdam = "SELECT population FROM cities WHERE name ='Rotterdam'"
+filtering(populationOfRotterdam);
+
+// 5-3 What's the top 10 countries based on surface area?
+
+let topTenCountries = "SELECT name FROM countries ORDER BY surface_area DESC LIMIT 10 ";
+filtering(topTenCountries);
+
+// 5-4 What's the top 10 cities with the highest population?
+let highestPopulation = "SELECT name FROM cities ORDER BY population DESC LIMIT 10 "
+filtering(highestPopulation);
+// 5-5 What's the population of the world ?
+let populationOfTheWorld = " SELECT SUM(DISTINCT population) FROM countries"
```

Comment:
Are you sure you want to use distinct here? If two countries have the same population number you just want to count it once :)? I don't think so. Distinct will count the same values only once.

Reply:
yes, I used SUM(population) but I was curious to know how Distinct works. and also forgot to delete it :)